orchestra/static/orchestra/task/js/controllers.js

```diff
@@ -116,28 +116,33 @@
       vm.saving = true;
       vm.saveError = false;
       vm.cancelAutoSave();
+      orchestraService.signals.save.before();
       $http.post('/orchestra/api/interface/save_task_assignment/',
                  {'task_id': vm.taskId, 'task_data': vm.taskAssignment.task.data})
         .success(function(data, status, headers, config) {
           vm.lastSaved = Date.now();
           // Reset timeout counter on save success
           vm.autoSaveTimeout = 10000;
+          orchestraService.signals.save.success();
         })
         .error(function(data, status, headers, config) {
           vm.saveError = true;
+          orchestraService.signals.save.error();
         })
         .finally(function() {
-          vm.saving = false;
           if (vm.saveError) {
             // Retry save with exp backoff
             vm.autoSaveTimeout *= 2;
             vm.scheduleAutoSave();
           }
+          orchestraService.signals.save.finally();
+          vm.saving = false;
```

Comment:
Originally wanted the `saving` flag to be reset at the very end, but it probably makes sense to change it before the next save is scheduled if there's an error!